src/track/beatgrid.cpp

```diff
@@ -147,9 +147,27 @@ audio::FramePos BeatGrid::findNthBeat(audio::FramePos position, int n) const {
         return audio::kInvalidFramePos;
     }
 
-    const double beatFraction = (position - firstBeatPosition()) / m_beatLengthFrames;
-    const double prevBeat = floor(beatFraction);
-    const double nextBeat = ceil(beatFraction);
+    double beatFraction = (position - firstBeatPosition()) / m_beatLengthFrames;
+    double prevBeat = floor(beatFraction);
+    double nextBeat = ceil(beatFraction);
+
+    // If the position is within 1/100th of the next or previous beat, treat it
+    // as if it is that beat.
```

Comment:
Please elaborate in the comment explaining this is required due to the imprecision of floating points.